client/js/src/client.ts

```diff
@@ -821,84 +822,129 @@ export function api_factory(
-											fn_index
-										});
+										// TODO: Find out how to print this information
+										// only during testing
+										// console.info("data", type, status, data);
 
-										if (complete) {
+										if (type == "heartbeat") {
+											return;
+										}
+
+										if (type === "update" && status && !complete) {
+											// call 'status' listeners
 											fire_event({
 												type: "status",
+												endpoint: _endpoint,
+												fn_index,
 												time: new Date(),
-												...complete,
+												...status
+											});
+										} else if (type === "complete") {
+											complete = status;
+										} else if (type == "unexpected_error") {
+											console.error("Unexpected error", status.message);
+											fire_event({
+												type: "status",
+												stage: "error",
+												message: "An Unexpected Error Occurred!",
+												queue: true,
+												endpoint: _endpoint,
+												fn_index,
+												time: new Date()
+											});
+										} else if (type === "log") {
+											fire_event({
+												type: "log",
+												log: data.log,
+												level: data.level,
+												endpoint: _endpoint,
+												fn_index
+											});
+										} else if (type === "generating") {
+											fire_event({
+												type: "status",
+												time: new Date(),
+												...status,
 												stage: status?.stage!,
 												queue: true,
 												endpoint: _endpoint,
 												fn_index
 											});
 										}
-									}
+										if (data) {
+											fire_event({
+												type: "data",
+												time: new Date(),
+												data: transform_files
+													? transform_output(
+															data.data,
+															api_info,
+															config.root,
+															config.root_url
+													  )
+													: data.data,
+												endpoint: _endpoint,
+												fn_index
+											});
+
+											if (complete) {
+												fire_event({
+													type: "status",
+													time: new Date(),
+													...complete,
+													stage: status?.stage!,
+													queue: true,
+													endpoint: _endpoint,
+													fn_index
+												});
+											}
+										}
 
-									if (status.stage === "complete" || status.stage === "error") {
-										if (event_callbacks[event_id]) {
-											delete event_callbacks[event_id];
-											if (Object.keys(event_callbacks).length === 0) {
-												close_stream();
+										if (
+											status.stage === "complete" ||
+											status.stage === "error"
+										) {
+											if (event_callbacks[event_id]) {
+												delete event_callbacks[event_id];
+												if (Object.keys(event_callbacks).length === 0) {
+													close_stream();
+												}
 											}
 										}
+									} catch (e) {
+										console.error("Unexpected client exception", e);
```

Comment:
This should fix the `HTTPException` errors we keep seeing in the logs.

### Explanation
https://www.loom.com/share/5f7abfdb2f2143c58982f11dab2538b2?sid=d176fd1c-b5f5-43f5-be5f-f72af9b3abea

### Fix
https://www.loom.com/share/d6b9d87d05b64f69bfb8158ac565593f?sid=ca833ba1-4579-4536-8e14-50396ba3261c